src/coprocessor/dag/expr/builtin_op.rs

```diff
@@ -131,7 +131,21 @@ impl ScalarFunc {
 
     pub fn time_is_null(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
         let arg = self.children[0].eval_time(ctx, row)?;
-        Ok(Some(arg.is_none() as i64))
+        if arg.is_none() {
+            return Ok(Some(true as i64));
+        }
+
+        if self.implicit_args.len() > 0 {
+            let is_not_null_col = self.implicit_args[0].i64();
+            if is_not_null_col > 0 && arg.unwrap().is_zero() {
```

Comment:
I think `is_not_null_col` is already pushed down?

Reply:
It should be `FieldTypeFlag::NOT_NULL`

Reply:
> I think `is_not_null_col` is already pushed down?

No, it is pushed down in this PR https://github.com/pingcap/tidb/pull/10957.